diff --git a/code/src/rss_reader/CMakeLists.txt b/code/src/rss_reader/CMakeLists.txt
index aed7201..7b2922c 100644
--- a/code/src/rss_reader/CMakeLists.txt
+++ b/code/src/rss_reader/CMakeLists.txt
@@ -14,8 +14,8 @@ SET(RESOURCE_SRCS rss_images.qrc)
 QT4_ADD_RESOURCES(IMAGES_RES ${RESOURCE_SRCS})
 
 # EXE
-ADD_EXECUTABLE(rss_reader.oar ${SRCS} ${IMAGES_RES})
-TARGET_LINK_LIBRARIES(rss_reader.oar onyx_cms onyx_ui onyx_wireless onyx_sys onyx_screen
+ADD_EXECUTABLE(rss_reader ${SRCS} ${IMAGES_RES})
+TARGET_LINK_LIBRARIES(rss_reader onyx_cms onyx_ui onyx_wireless onyx_sys onyx_screen
   ${QT_LIBRARIES}
   ${ADD_LIB}
 )
