diff -ruN onyx-intl-booxsdk-a0c3c9a/code/include/onyx/ui/status_bar.h onyx-intl-booxsdk-a0c3c9a.obx/code/include/onyx/ui/status_bar.h
--- onyx-intl-booxsdk-a0c3c9a/code/include/onyx/ui/status_bar.h	2011-04-18 06:54:03.000000000 +0200
+++ onyx-intl-booxsdk-a0c3c9a.obx/code/include/onyx/ui/status_bar.h	2011-05-24 19:29:18.520396072 +0200
@@ -37,6 +37,7 @@
     bool setProgress(const int value, const int total,
             bool show_message = true, const QString &message = "");
     bool setMessage(const QString & message);
+    bool setMessage(const QString & message, const QFont & font);
     bool setItemState(const StatusBarItemType type, const int state);
 
     void enableJumpToPage(bool enable) { enable_jump_to_page_ = enable; }
diff -ruN onyx-intl-booxsdk-a0c3c9a/code/include/onyx/ui/ui_global.h onyx-intl-booxsdk-a0c3c9a.obx/code/include/onyx/ui/ui_global.h
--- onyx-intl-booxsdk-a0c3c9a/code/include/onyx/ui/ui_global.h	2011-04-18 06:54:03.000000000 +0200
+++ onyx-intl-booxsdk-a0c3c9a.obx/code/include/onyx/ui/ui_global.h	2011-05-24 19:29:18.520396072 +0200
@@ -167,6 +167,9 @@
     ABOUT_INFO,
     FULL_SCREEN,
     EXIT_FULL_SCREEN,
+    REMOVE_SD,
+    STANDBY,
+    SHUTDOWN,
 };
 
 enum AnnotationMode
diff -ruN onyx-intl-booxsdk-a0c3c9a/code/src/ui/onyx_ui_images.qrc onyx-intl-booxsdk-a0c3c9a.obx/code/src/ui/onyx_ui_images.qrc
--- onyx-intl-booxsdk-a0c3c9a/code/src/ui/onyx_ui_images.qrc	2011-04-18 06:54:03.000000000 +0200
+++ onyx-intl-booxsdk-a0c3c9a.obx/code/src/ui/onyx_ui_images.qrc	2011-05-24 19:32:46.330398881 +0200
@@ -231,5 +231,9 @@
     <file>images/locale.png</file>
     <file>images/locale_small.png</file>
     <file>images/enter_key.png</file>
+
+    <file>images/remove_sd.png</file>
+    <file>images/shutdown.png</file>
+    <file>images/standby.png</file>
   </qresource>
 </RCC>
diff -ruN onyx-intl-booxsdk-a0c3c9a/code/src/ui/status_bar.cpp onyx-intl-booxsdk-a0c3c9a.obx/code/src/ui/status_bar.cpp
--- onyx-intl-booxsdk-a0c3c9a/code/src/ui/status_bar.cpp	2011-04-18 06:54:03.000000000 +0200
+++ onyx-intl-booxsdk-a0c3c9a.obx/code/src/ui/status_bar.cpp	2011-05-24 19:29:18.523396037 +0200
@@ -146,6 +146,19 @@
     return false;
 }
 
+bool StatusBar::setMessage(const QString & message, const QFont & font)
+{
+    StatusBarItem *ptr = item(MESSAGE, false);
+    if (ptr)
+    {
+        StatusBarItemMessage *text= static_cast<StatusBarItemMessage*>(ptr);
+        text->setFont(font);
+        text->setMessage(message);
+        return true;
+    }
+    return false;
+}
+
 void StatusBar::showItem(StatusBarItemType id, bool show)
 {
     StatusBarItem *ptr = item(id, false);
diff -ruN onyx-intl-booxsdk-a0c3c9a/code/src/ui/system_actions.cpp onyx-intl-booxsdk-a0c3c9a.obx/code/src/ui/system_actions.cpp
--- onyx-intl-booxsdk-a0c3c9a/code/src/ui/system_actions.cpp	2011-04-18 06:54:03.000000000 +0200
+++ onyx-intl-booxsdk-a0c3c9a.obx/code/src/ui/system_actions.cpp	2011-05-24 19:29:18.524396027 +0200
@@ -112,6 +112,36 @@
                 actions_.push_back(close);
                 break;
             }
+        case REMOVE_SD:
+            {
+                shared_ptr<QAction> removeSd(new QAction(exclusiveGroup()));
+                removeSd->setCheckable(true);
+                removeSd->setText(QCoreApplication::tr("Safely Remove SD"));
+                removeSd->setIcon(QIcon(QPixmap(":/images/remove_sd.png")));
+                removeSd->setData(REMOVE_SD);
+                actions_.push_back(removeSd);
+                break;
+            }
+        case STANDBY:
+            {
+                shared_ptr<QAction> standby(new QAction(exclusiveGroup()));
+                standby->setCheckable(true);
+                standby->setText(QCoreApplication::tr("Standby"));
+                standby->setIcon(QIcon(QPixmap(":/images/standby.png")));
+                standby->setData(STANDBY);
+                actions_.push_back(standby);
+                break;
+            }
+        case SHUTDOWN:
+            {
+                shared_ptr<QAction> shutdown(new QAction(exclusiveGroup()));
+                shutdown->setCheckable(true);
+                shutdown->setText(QCoreApplication::tr("Shutdown"));
+                shutdown->setIcon(QIcon(QPixmap(":/images/shutdown.png")));
+                shutdown->setData(SHUTDOWN);
+                actions_.push_back(shutdown);
+                break;
+            }
         }
     }
 }
